File: kensu-spark-example/script_with_kensu.scala
import org.apache.spark.sql.SparkSession

// P.S. make sure you have downloaded the jar below
val spark = SparkSession
  .builder()
  .appName("Example")
  .config("spark.driver.extraClassPath", "kensu-spark-collector-1.0.0_spark-3.0.1.jar")
  .getOrCreate()


implicit val ch = new io.kensu.dodd.sdk.ConnectHelper("conf.ini")
io.kensu.sparkcollector.KensuSparkCollector.KensuSparkSession(spark).track(ch.properties.get("kensu_ingestion_url").map(_.toString), None)(ch.properties.toList:_*)

val business_info = spark.read.option("inferSchema","true").option("header","true").csv("data/business-data.csv")
val customers_info = spark.read.option("inferSchema","true").option("header","true").csv("data/customers-data.csv")
val contact_info = spark.read.option("inferSchema","true").option("header","true").csv("data/contact-data.csv")

val df = business_info.join(customers_info,business_info("id") === customers_info("id"), "inner").drop(customers_info("id"))
val df_final = df.join(contact_info,df("id") === contact_info("id"), "inner").drop(contact_info("id"))

df_final.write.mode("overwrite").save("data/data")

File: kensu-spark-example/script_with_kensu.py
from pyspark.sql import SparkSession
from kensu.pyspark import init_kensu_spark

# download kensu spark collector jar if doesn't exist in ../lib/
from spark_collector_downloader import maybe_download_spark_collector, kensu_agent_jar_local_path
maybe_download_spark_collector(kensu_agent_jar_local_path)


#Add the path to the .jar to the SparkSession
spark = SparkSession.builder.appName("Example")\
    .config("spark.driver.extraClassPath", kensu_agent_jar_local_path)\
    .getOrCreate()

#Init Kensu

init_kensu_spark(spark)


business_info = spark.read.option("inferSchema","true").option("header","true").csv("data/business-data.csv")
customers_info = spark.read.option("inferSchema","true").option("header","true").csv("data/customers-data.csv")
contact_info = spark.read.option("inferSchema","true").option("header","true").csv("data/contact-data.csv")


df = business_info.join(customers_info,['id']).join(contact_info,['id'])

df.write.mode("overwrite").save("data/data")